var/spack/repos/builtin/packages/siesta/package.py

```diff
@@ -5,16 +5,38 @@
 
 
 import os
+import shutil
 
+from spack.build_systems import cmake
 from spack.package import *
 
 
-class Siesta(MakefilePackage):
+class Siesta(MakefilePackage, CMakePackage):
     """SIESTA performs electronic structure calculations and ab initio molecular
     dynamics simulations of molecules and solids.
     """
 
+    build_system(conditional("cmake", when="@5.0.0:"), "makefile", default="makefile")
+
     homepage = "https://departments.icmab.es/leem/siesta/"
+    git = "https://gitlab.com/siesta-project/siesta"
+
+    version(
+        "5.0.1",
+        url="https://gitlab.com/siesta-project/siesta/-/archive/5.0.1/siesta-5.0.1.tar.gz",
+        sha256="800a22a831c1d36c6f5fe4aa9c49ee510cbd49a0b2f87b3c8bf3edb6ebd0193a",
+    )
+
+    version(
+        "5.0.0",
+        url="https://gitlab.com/siesta-project/siesta/-/archive/rel-5.0/siesta-rel-5.0.tar.gz",
+        sha256="a407c31e8fd101efe4bfc68831870c524a8c397a39de7271d31ce0e6a4df92bb",
+    )
+    version(
+        "4.1.5",
+        url="https://gitlab.com/siesta-project/siesta/-/archive/v4.1.5/siesta-v4.1.5.tar.gz",
+        sha256="518df31aa6213af5e24cc73abb537b2c89a925b487171f5339d743d0c7140b3f",
+    )
```

Comment:
Good, but the sha265 of 5.0 also changed meanwhile:
```py
         "5.0.0",
         url="https://gitlab.com/siesta-project/siesta/-/archive/rel-5.0/siesta-rel-5.0.tar.gz",
-        sha256="a407c31e8fd101efe4bfc68831870c524a8c397a39de7271d31ce0e6a4df92bb",
+        sha256="0b40c341dfd47e99e7e191189600bbcaadb0f9af03977fefed6a69836bd523e4",
```
Will the developers refrain from retrospectively re-uploading the tarballs with different content in the future? I ask because doing so is not compatible with using `spack`.

Reply:
I agree with you. I hope it won't be changed as they have already communicated the versions to the Siesta community.